### PR TITLE
Crypto refactor nonce padding

### PR DESCRIPTION
**Pull request by folkertdev**
if we allow it, we should test it

### CODE REVIEW

File: ntp-proto/src/packet/extensionfields.rs [hunk 1]
```diff
@@ -300,27 +297,39 @@ impl<'a> ExtensionField<'a> {
...
+        let nonce_padding = padded_nonce_length - nonce_length;
+        cur_extension_field[header_size + nonce_length..][..nonce_padding]
+            .copy_from_slice(&padding[..nonce_padding]);
+
+        // zero out the ciphertext padding
+        let ciphertext_padding = padded_ciphertext_length - ciphertext_length;
+        debug_assert_eq!(
+            ciphertext_padding, 0,
+            "extension field encoding should add padding"
+        );
```
Comment (folkertdev) [inline ntp-proto/src/packet/extensionfields.rs:317]: I believe this just should always be zero in practice. We don't test with a `ciphertext_padding > 0` at the moment, and I don't think it's possible to create with our logic a plaintext that would require ciphertext padding

File: ntp-proto/src/packet/extensionfields.rs [hunk 2]
```diff
@@ -549,10 +562,6 @@ impl<'a> RawEncryptedField<'a> {
         let nonce_length = u16::from_be_bytes([b0, b1]) as usize;
         let ciphertext_length = u16::from_be_bytes([b2, b3]) as usize;
 
-        if nonce_length != 16 {
-            return Err(IncorrectLength);
-        }
-
```
Comment (folkertdev) [inline ntp-proto/src/packet/extensionfields.rs:555]: this constraint is now no longer required, and an implementation detail of the ciphers that we provide today.

**Comment by rnijveld** [comment]
The code LGTM, but I'm not enough into crypto to have anything else to say about this, maybe @davidv1992 can take a look as well?

File: ntp-proto/src/packet/extensionfields.rs (hunk 1 above)
Comment (davidv1992) [reply ntp-proto/src/packet/extensionfields.rs:317]: Given the multiple-of-4 requirement even on encrypted efs you might be right. I am comfortable with this debug assert for now.

**Pull request merged**